From 6a04f228d63724949db9f4767f7092e430e039e5 Mon Sep 17 00:00:00 2001
Message-ID: <6a04f228d63724949db9f4767f7092e430e039e5.1711239854.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1711239854.git.sharlatanus@gmail.com>
References: <cover.1711239854.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Fri, 22 Mar 2024 22:11:19 +0000
Subject: [PATCH 07/12] gnu: Add go-github-com-xhit-go-str2duration-v2.

* gnu/packages/golang-xyz.scm (go-github-com-xhit-go-str2duration-v2): New variable.

Change-Id: I30425e498db6dd8cb736eedffe006346973628ed
---
 gnu/packages/golang-xyz.scm | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/gnu/packages/golang-xyz.scm b/gnu/packages/golang-xyz.scm
index e012beabaf..e3c2d86062 100644
--- a/gnu/packages/golang-xyz.scm
+++ b/gnu/packages/golang-xyz.scm
@@ -2035,6 +2035,31 @@ (define-public go-github-com-whyrusleeping-go-sysinfo
 @code{MemoryInfo}.")
     (license license:expat)))
 
+(define-public go-github-com-xhit-go-str2duration-v2
+  (package
+    (name "go-github-com-xhit-go-str2duration-v2")
+    (version "2.1.0")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/xhit/go-str2duration")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "1c9zi9mfy5ww413y1jpfh1rdis43lvd5v6gvajqzh4q1km9lyxjj"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:import-path "github.com/xhit/go-str2duration/v2"))
+    (home-page "https://github.com/xhit/go-str2duration")
+    (synopsis "Convert string to duration in golang")
+    (description
+     "This package allows to get a @code{time.Duration} from a string.  The
+string can be a string retorned for @code{time.Duration} or a similar string
+with weeks or days too.")
+    (license license:bsd-3)))
+
 (define-public go-go-uber-org-automaxprocs
   (package
     (name "go-go-uber-org-automaxprocs")
-- 
2.41.0

